workos/events.py

```diff
@@ -76,22 +85,14 @@ def list_events(
             params=params,
             token=workos.api_key,
         )
-
-        if "default_limit" in locals():
-            if "metadata" in response and "params" in response["metadata"]:
-                response["metadata"]["params"]["default_limit"] = default_limit
-            else:
-                response["metadata"] = {"params": {"default_limit": default_limit}}
-
-        response["metadata"] = {
-            "params": params,
-            "method": Events.list_events,
-        }
-
-        return response
+        return WorkOsListResource(
+            list_method=self.list_events,
+            list_args=params,
+            **ListPage[Event](**response).model_dump(exclude_unset=True),
```

Comment:
I had to use `exclude_unset=True` otherwise this would always serialize `before` in the `list_metadata`. I think it's because the resource can't tell the difference between `ListAfterMetadata` and `ListMetadata`.

The other option is to and a third generic parameter and explicitly pass in the `ListMetadata` shape. But, since python type hint generics don't set support default values for generics, every list resource would have to pass in the list metadata type.

Reply:
I realized there is another option. We could have separate `WorkOsForwardListResource` and `WorkOsBidirectionalListResource`.

Not sure how much benefit that will bring. It would almost definitely be more code.